### PR TITLE
Update Icon Support

### PR DESCRIPTION
**Pull request by joshmedeski**
Closes https://github.com/jesseduffield/lazygit/issues/1933

I'm adding additional Nerd font icon support:

- Files status
- Git status

Preview:
Screen Shot 2022-05-09 at 2 50 49 PM

### CODE REVIEW

File: pkg/gui/presentation/icons/git_icons.go [hunk 1]
```diff
@@ -59,3 +60,24 @@ func IconForRemote(remote *models.Remote) string {
...
 }
+
+
+func IconForChangeStatus(changeStatus string) string {
+	nfOctDiffAdded    := "\uf457 " // 
+	nfOctDiffModified := "\uf459 " // 
+	nfOctDiffRemoved  := "\uf458 " // 
+	nfOctDiffRenamed  := "\uf45a " // 
+
+	switch changeStatus {
```
Comment (Ryooooooga) [inline pkg/gui/presentation/icons/git_icons.go:71]: Conflicts are not handled.

Comment (Ryooooooga) [inline pkg/gui/presentation/icons/git_icons.go:71]: Conflicts are not handled.

Comment (joshmedeski) [reply pkg/gui/presentation/icons/git_icons.go:71]: Good point. I think there's lots more areas I need to cover before we merge this PR.

**Comment by danielebra** [comment]
Great idea!
Have you considered emoji support for [gitmoji](https://gitmoji.dev/) also?

**Comment by joshmedeski** [comment]
I've improved the code, added some more features, and have decent fallbacks.

@jesseduffield can you (or anyone else) do some testing and let me know if anything else needs to be done?

File: pkg/gui/presentation/files.go [hunk 2]
```diff
@@ -86,12 +81,12 @@ func renderAux(
 	}
 
 	if collapsedPaths.IsCollapsed(s.GetPath()) {
-		return []string{prefix + COLLAPSED_ARROW + " " + renderLine(s, depth)}
+		return []string{prefix + icons.EXPANDED_DIR_ARROW_ICON + " " + renderLine(s, depth)}
```
Comment (Ryooooooga) [inline pkg/gui/presentation/files.go:84]: They are used even if Nerdfonts is not enabled.

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Apologies for the long wait, I've left some feedback. One of the things that needs more thought is the file icons. Specifically: when we're showing both the staged and unstaged change at the same time it's fairly easy to see how files line up, but when we omit one of the characters (as this PR does) it becomes harder to see at a glance whether the present character represents a staged or unstaged change (for example if a file within a folder is above a file from the parent folder).

One way to get around this is to have all the file change characters on the very left of the panel, no matter how nested the file itself is. Trying this out locally though I don't really like how it looks.
image

There's definitely room for improvement on how we render the files panel but I think it's gonna need some more consideration before we commit to anything

**Comment by joshmedeski** [comment]
Thanks for all the feedback. It looks like some of my opinionated changes should be reverted for now and individually introduced in separate PRs. I have a thought for how to do it more gracefully.

I'll make some updates in the next week and flag you for code review again.

**Comment by joshmedeski** [comment]
I'm going to close this for now. It seems like some of my breaking changes shouldn't all happen in one big PR, plus some of the icons I chose aren't preferred by the reviewers.

I'll keep maintaining my branch and I can change it into smaller PRs for more bite-sized changes.

**Comment by joshmedeski** [comment]
Hey, I'm ready to revisit this work and get some of the changes merged. What's the status of the big refactor? I'm happy to add full Nerd Font support.

**Pull request closed without merging**